nltk/sem/relextract.py

```diff
@@ -198,17 +198,20 @@ def extract_rels(subjclass, objclass, doc, corpus='ace', pattern=None, window=10
     :return: see ``mk_reldicts``
     :rtype: list(defaultdict)
     """
-
-    if subjclass and subjclass not in NE_CLASSES[corpus]:
-        if _expand(subjclass) in NE_CLASSES[corpus]:
-            subjclass = _expand(subjclass)
-        else:
-            raise ValueError("your value for the subject type has not been recognized: %s" % subjclass)
-    if objclass and objclass not in NE_CLASSES[corpus]:
-        if _expand(objclass) in NE_CLASSES[corpus]:
-            objclass = _expand(objclass)
-        else:
-            raise ValueError("your value for the object type has not been recognized: %s" % objclass)
+    
+    for i, s in enumerate(subjclass):
```

Comment:
@helshowk, consider this code,
```python
# check whether 3 is in a list
3 in [1, 2, 4, 5, 3]
```
it will iterate over the list to figure out whether there is `3` or not. In the worst case scenario it will make as many comparison checks as there are in the list.

If a set is build, then the check
```python
3 in set([1, 2, 4, 5, 3])
```
in the worst case will take less comparisons than there are elements in the list, thus the check will be sublinear.